features/support/env.rb

```diff
@@ -3,14 +3,28 @@
 require 'open4'
 require 'rspec'
 
+
 SOURCE_DIRECTORY = "#{File.dirname(__FILE__)}/../../src"
 SHELL_OVERRIDE_DIRECTORY = "#{File.dirname(__FILE__)}/shell_overrides"
 
+MEMOIZED_REPOSITORY_BASE = Dir.mktmpdir 'memoized'
 REPOSITORY_BASE = Dir.mktmpdir
 TOOLS_INSTALLED_FILENAME = "#{REPOSITORY_BASE}/tools_installed.txt"
 
-Before do
-  Dir.chdir REPOSITORY_BASE
+
+# load memoized environment by copying contents
+# of MEMOIZED_REPOSITORY_BASE to REPOSITORY_BASE
+def setup_environment
+  FileUtils.rm_rf Dir.glob("#{REPOSITORY_BASE}/*")
```

Comment:
I just tested it and cp -r created the destination directory if it doesn't exist.